dind/start.sh

```diff
@@ -95,8 +95,44 @@ start_master ()
   docker tag k8s.gcr.io/kube-proxy:$(cat /docker_version) k8s.gcr.io/kube-proxy-amd64:$(cat /docker_version)
   docker tag k8s.gcr.io/kube-scheduler:$(cat /docker_version) k8s.gcr.io/kube-scheduler-amd64:$(cat /docker_version)
 
+  cat <<EOPOLICY > /etc/kubernetes/audit-policy.yaml
+apiVersion: audit.k8s.io/v1beta1
+kind: Policy
+omitStages:
+  - "RequestReceived"
+rules:
+- level: RequestResponse
+  resources:
+  - group: "" # core
+    resources: ["pods", "secrets"]
+  - group: "extensions"
+    resources: ["deployments"]
+EOPOLICY
+  chmod 0600 /etc/kubernetes/audit-policy.yaml
+  cat <<EOF > /etc/kubernetes/kubeadm.conf
+# Only adding items not in 'kubeadm config print-default'
+apiVersion: kubeadm.k8s.io/v1alpha3
+kind: InitConfiguration
+kubernetesVersion: $(cat source_version | sed 's/^.//')
+auditPolicy:
+  path: /etc/kubernetes/audit-policy.yaml
+featureGates:
+  Auditing: true
+networking:
+  podSubnet: 192.168.0.0/16
+bootstrapTokens:
+- groups:
+  - system:bootstrappers:kubeadm:default-node-token
+  token: abcdef.abcdefghijklmnop
```

Comment:
This does result it:

```
You can now join any number of machines by running the following on each node
as root:

  kubeadm join 172.18.0.2:6443 --token abcdef.abcdefghijklmnop --discovery-token-ca-cert-hash sha256:008789ee5ec6758715f39fda15406615c0d7150eb386e5b794cdd066640d46a2
```

However on my minion nodes:

```
[discovery] Created cluster-info discovery client, requesting info from "https://172.18.0.2:6443"
[discovery] Failed to connect to API Server "172.18.0.2:6443": token id "abcdef" is invalid for this cluster or it has expired. Use "kubeadm token create" on the master node to creating a new valid token
[discovery] abort connecting to API servers after timeout of 5m0s
```
I'm unsure if this is related to the certs:

```
E0816 23:30:21.344248       1 memcache.go:134] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
I0816 23:30:31.338437       1 logs.go:50] http: TLS handshake error from 172.17.0.1:59826: remote error: tls: bad certificate
I0816 23:30:41.339545       1 logs.go:50] http: TLS handshake error from 172.17.0.1:59838: remote error: tls: bad certificate
I0816 23:30:51.335838       1 logs.go:50] http: TLS handshake error from 172.17.0.1:59850: remote error: tls: bad certificate
```

Reply:
This issue went away when we added apiServerCertSANs in this way:

```yaml
apiServerCertSANs:
$(echo $1 | sed -e 's: :\n:g' | sed 's:^:- :') 
- kubernetes 
```